frontend/src/components/WaitingRoom/ControlPanel/ControlPanel.tsx

```
@@ -54,48 +55,89 @@ const ControlPanel = ({
   openAudioOutput,
   anchorEl,
 }: ControlPanelProps): ReactElement | false => {
+  const [openMoreOptions, setOpenMoreOptions] = useState(false);
+  const [moreOptionsAnchorEl, setMoreOptionsAnchorEl] = useState<HTMLElement | null>(null);
+  const handleCloseMoreOptions = () => {
+    setOpenMoreOptions(false);
+    setMoreOptionsAnchorEl(null);
+  };
+  const handleOpenMoreOptions = (event: MouseEvent<HTMLElement>) => {
+    setMoreOptionsAnchorEl(event.currentTarget);
+    setOpenMoreOptions(true);
+  };
+
   const { t } = useTranslation();
   const isSmallViewport = useIsSmallViewport();
   const { allMediaDevices } = useDevices();
   const { localAudioSource, localVideoSource, changeAudioSource, changeVideoSource } =
     usePreviewPublisherContext();
   const { currentAudioOutputDevice, setAudioOutputDevice } = useAudioOutputContext();
+  const theme = useCustomTheme();
 
   const allowDeviceSelection = useAppConfig(
     ({ waitingRoomSettings }) => waitingRoomSettings.allowDeviceSelection
   );
 
   const buttonSx: SxProps = {
-    borderRadius: '10px',
-    color: 'rgb(95, 99, 104)',
-    textTransform: 'none', // ensures that the text is not upper case
-    border: 'none',
-    boxShadow: 'none',
-    whiteSpace: 'nowrap',
+    display: 'flex',
+    alignItems: 'center',
+    gap: 1.5,
+    borderRadius: theme.shapes.borderRadiusMedium,
+    padding: '8px',
+    color: theme.colors.textSecondary,
+    fontWeight: theme.typography.weight.caption.value,
+    fontSize: theme.typography.typeScale.desktop['body-base'].fontSize.value,
+
+    minWidth: 0,
+
     '&:hover': {
-      border: 'none',
-      boxShadow: 'none',
+      backgroundColor: theme.colors.background,
     },
   };
 
+  const textSx: SxProps = {
+    flex: '1 1 0',
+    minWidth: 0,
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  };
+
   return (
     allowDeviceSelection && (
```

Since we are doing some updates, let’s move allowDeviceSelection to WaitingRoom. This kind of component, which may or may not render anything, is a React anti-pattern and is difficult to maintain.

```tsx
 {allowDeviceSelection && accessStatus === DEVICE_ACCESS_STATUS.ACCEPTED && (
    <ControlPanel
      handleAudioInputOpen={handleAudioInputOpen}
      handleVideoInputOpen={handleVideoInputOpen}
      handleAudioOutputOpen={handleAudioOutputOpen}
      handleClose={handleClose}
      openAudioInput={openAudioInput}
      openVideoInput={openVideoInput}
      openAudioOutput={openAudioOutput}
      anchorEl={anchorEl}
    />
  )}
```

Each parent component should be the orchestrator of its children, instead of the children silently hidden conditionally.